### DISCUSSION TITLE
What is the best way to search an indexed column via JS library?

### DISCUSSION THREAD

**Discussion by akiarostami** [pocketbase/pocketbase#4404] [Q&A]
Hello,

I'm sure there should be a way to search an index column, but somehow I can't find it. All I find is the general search like: `pb.collection(collectionIdOrName).getFirstListItem(filter, options = {});` 

Any help with this is highly appreciated!

**Comment by ganigeorgiev** [comment]
Why the `filter` parameter doesn't work in your case?

Here is one example:

```js
// or if you want to bind parameters from a map
// await pb.collection("example").getFirstListItem(pb.filter("name={:name}", {name: "test"});
await pb.collection("example").getFirstListItem("name = 'test'");
```
_(you can also find JS SDK code samples in the Admin UI API Preview panel for the specific collection)_

**Reply by akiarostami** [reply]
Yes, this is what I'm doing right now and it works. Sorry if I wasn't clear. I just wasn't sure if this automatically uses the index, or if there is another method to use an index for search, since I'm searching a unique and indexed field.

**Reply by ganigeorgiev** [reply]
There is no method available with the web APIs that will force/hint the use of an index.

Whether the SQLite query planner will use the index depends on the other used filters/sort parameters (you can inspect the resulting query with `--dev`).

It is possible to enforce an index with the [INDEXED BY](https://www.sqlite.org/lang_indexedby.html) statement but you'll need to write a custom raw query for that.

**Reply by akiarostami** [reply]
Thank you. Here's my real-case scenario: I have a table of articles with a "slut" column which is indexed and unique.  If I use:

```js
await pb.collection("articles").getFirstListItem("name = 'slut'");
```
would it use the index for the search?

**Reply by ganigeorgiev** [reply]
> would it use the index for the search?

If I have to guess for this particular case, then Yes.

But again whether the SQLite query planner will use the index depend on the resulting SQL query.
The resulting query could be affected by the API rules, `filter` and `sort` parameters because everything is converted into a single SQL statement (you can explore it with `--dev`).
Keep in mind that forcing a particular index may not always be the most efficient way to resolve the query and for most of the cases I'd recommend to leave the decision how a query will be resolved and what indexes to use to the query planner.
